funsor/adjoint.py

```diff
@@ -67,10 +67,7 @@ def __enter__(self):
         self._old_interpretation = interpreter.get_interpretation()
         return super().__enter__()
 
-    def adjoint(self, sum_op, bin_op, root, targets=None, *, batch_vars=frozenset()):
-        # TODO Replace this with root + Constant(...) after #548 merges.
-        root_vars = root.input_vars | batch_vars
-
+    def adjoint(self, sum_op, bin_op, root, targets=None, *, batch_vars=set()):
```

Comment:
note that `batch_vars` can now change during the course of `adjoint`, e.g. in Precondition where the aux vars aren't know until each `Approximate` term is hit.